pkg/config/expand.go

```diff
@@ -339,6 +339,9 @@ func validateModuleSettingReference(bp Blueprint, mod Module, r Reference) error
 	}
 
 	if err := validateModuleReference(bp, mod, r.Module); err != nil {
+		if _, ok := err.(InvalidModuleError); ok && bp.Vars.Has(r.Name) {
+			err = HintError{fmt.Sprintf("vars.%s", r.Name), err}
+		}
```

Comment:
Please add test case for `validateModuleSettingReference`

Reply:
Done.